### PR TITLE
Hooli-style Popup

### PR DESCRIPTION
**Pull request by pashapushak1996**
### Hooli-style Popup

[Demo](https://pashapushak1996.github.io/html-css-popup-kottans/) |
[Code base](https://github.com/pashapushak1996/html-css-popup-kottans)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Uchen&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="css/main.css">
+    <title>Popup</title>
+</head>
+<body>
+<div class="wrapper">
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:15]: Please make indentations less, to make it more readable

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,204 @@
+<!doctype html>
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:1]: Why from lowercase?

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                <li>
+                    <a href="#" class="header__link">Mail</a>
+                </li>
+                <li>
+                    <a href="#" class="header__link">Images</a>
+                </li>
+                <li>
+                    <input type="checkbox" id="app-icon" name="app-icon">
+                    <label for="app-icon" class="app-icon">
+                        <img src="icons/app-icon.png" alt="">
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:28]: `alt` attribute shouldn't be empty

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                    </label>
+                    <div class="popup">
+                        <div class="popup__inner">
+                            <ul class="popup__items">
+                                <li>
+                                    <a href="#" class="popup__item">
+                                        <div class="popup__image">
+                                            <img src="icons/calendar-icon.png" alt="calendar-icon">
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:36]: alt="calendar icon" will be better. Imagine that you can't see an image but need description to understand what's on it
https://www.w3schools.com/tags/att_img_alt.asp

**Review by kasionio — CHANGES_REQUESTED** [review]
Please remove redundant files from PR. We just need files which need to be reviewed here

File: submissions/pashapushak1996/html-css-popup/index.html (hunk 4 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/index.html]: I solve this. Thanks

File: submissions/pashapushak1996/html-css-popup/index.html (hunk 3 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/index.html]: I solve this. Thanks

File: submissions/pashapushak1996/html-css-popup/index.html (hunk 2 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/index.html]: I solve this. Thanks

File: submissions/pashapushak1996/html-css-popup/index.html (hunk 1 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/index.html:15]: I solve this. Thanks

File: submissions/pashapushak1996/html-css-popup/scss/main.scss [hunk 5]
```diff
@@ -0,0 +1,2 @@
+@import "base";
+@import "layout";
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/main.scss:2]: empty line needed

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+  }
+
+  &__item {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+
+    gap: 10px;
+
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:212]: no need to leave empty spaces here

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+  color: $grey-2;
+}
+
+.wrapper {
+  min-height: 100%;
+}
+
+.container {
+  max-width: 1200px;
+
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:14]: please remove unnecessary empty lines across the file

**Review by kasionio — CHANGES_REQUESTED** [review]
1. Seems top arrow has more thickness when popup border, should to be fixed
2. Please fix mobile view 
Pic 1 -  Screen Shot 2022-08-05 at 22 43 36


Pic 2- Screen Shot 2022-08-05 at 22 44 35

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss (hunk 7 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/scss/_layout.scss]: I did this.

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+  <meta name="viewport"
+        content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link href="https://fonts.googleapis.com/css2?family=Uchen&display=swap" rel="stylesheet">
+  <link rel="stylesheet" href="css/main.css">
+  <title>Popup</title>
+</head>
+<body>
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:14]: Please keep indentants. You can use [prettier](https://prettier.io/) in your IDE or even online

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+    width: 25px;
+    height: 25px;
+  }
+
+  &__profile {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:53]: Please remove all redundant spaces across this file. You don't need any spaces inside the CSS class

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+    height: 25px;
+  }
+
+}
+
+.popup {
+  display: none;
+  flex-direction: column;
+
+  width: 340px;
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:105]: Try to use relative units `rem` and `em`, it's a good practice nowadays. 
You can read articles [here](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units), [here](https://www.freecodecamp.org/news/what-is-rem-in-css/) or any related to it's info

**Review by kasionio — CHANGES_REQUESTED** [review]
The mobile view is still not fixed, please take a look. And look at the comments below

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss (hunk 10 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/html-css-popup/scss/_layout.scss]: I did this. Thanks for article!

**Comment by pashapushak1996** [comment]
I don't know how I can fix it. Do you help me?

**Comment by kasionio** [comment]
You can ask specific questions, or ask assistance in a chat, or take a look at works of other students. The choice is yours :)

**Comment by pashapushak1996** [comment]
I did this.

**Comment by kasionio** [comment]
Excuse me, did what?

**Comment by pashapushak1996** [comment]
I solved the problem with popup top arrow)

**Review by kasionio — CHANGES_REQUESTED** [review]
Fixed one, made another ))
Screen Shot 2022-08-10 at 11 43 23

**Comment by pashapushak1996** [comment]
But I watched the submissions and one had almost the same solution.

**Comment by pashapushak1996** [comment]
Maybe I Do this popup without Arrow😅

**Comment by kasionio** [comment]
Accept the challenge, it is not that difficult :) And struggling is also a process which developer facing pretty often, so it should be a good practice ;)

**Comment by pashapushak1996** [comment]
I solved this problem. Thank you for pushing me forward.

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+            <div class="popup__inner">
+              <ul class="popup__items">
+                <li>
+                  <a href="#" class="popup__item">
+                    <div class="popup__image">
+                      <img
+                          src="icons/calendar-icon.png"
+                          alt="calendar-icon" />
+                    </div>
+                    <p>Calendar</p>
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:41]: I missed it earlier, better not to use `p` inside `a`.
https://caninclude.glitch.me/caninclude?child=p&parent=a

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+  .popup__items_hidden {
+    display: none;
+    border-top: 2px solid $grey-1;
+  }
+
+  input:checked ~ .popup__items_hidden {
+    display: grid;
+  }
+
+  input:checked ~ label {
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:158]: I have the same properties here, why not to unite them?

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+
+  &__link_notify {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 2.5rem;
+    height: 2.5rem;
+  }
+
+  &__profile {
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:45]: this is a copy what we have above except one line, try do not duplicate your code.

File: submissions/pashapushak1996/html-css-popup/scss/_layout.scss [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 2.5rem;
+    height: 2.5rem;
+    border-radius: 50%;
+  }
+}
+
+.header__list li:nth-child(3) {
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/scss/_layout.scss:55]: why do you need such specific selector here?

File: submissions/pashapushak1996/html-css-popup/index.html [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+                    <div class="popup__image">
+                      <img
+                          src="icons/telegram-icon.png"
+                          alt="telegram-icon" />
+                    </div>
+                    <p>Telegram</p>
+                  </a>
+                </li>
+              </ul>
+              <p class="popup__text">Even more from cottans</p>
```
Comment (kasionio) [inline submissions/pashapushak1996/html-css-popup/index.html:199]: Kottans 😹

**Review by kasionio — CHANGES_REQUESTED** [review]
Nice job! Glad you succeed 😺 Look at the comments. We are almost done

**Review by kasionio — CHANGES_REQUESTED** [review]
Small question. 
Why do you prefer SCSS and not CSS in this task?

**Comment by pashapushak1996** [comment]
Thank you. I didn't know about extend and mixin)
I don't know why I used SCSS, maybe I wanted to seem more skilled

**Review by kasionio — CHANGES_REQUESTED** [review]
Please always add new line at the end of the file. You can set it in your IDE settings, not to do it manually
Screen Shot 2022-08-12 at 19 58 16

https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

You use tools, but not use it's advantages. Try to do exercises by yourself, explore materials and if something unclear you can ask, and then see works of other students, in this order 😄 The main goal is to understand the material and use it in practice.

**Comment by pashapushak1996** [comment]
I added empty lines to files. Thank you for review and advices!!!

**Comment by pashapushak1996** [comment]
I added empty lines to file. Thank you for review and advices!

**Review by kasionio — APPROVED** [review]

**Pull request merged**